File: main/java/com/marsrovers/io/IOHandler.java
package com.marsrovers.io;


import java.util.Scanner;

public class IOHandler {

    private final Scanner scanner;
    private final Output output;

    IOHandler(Scanner scanner, Output output) {
        this.scanner = scanner;
        this.output = output;
    }

    public String readWithMessage(String readingMessage) {
        output.write(readingMessage);
        return scanner.nextLine();
    }

    public void write(String message) {
        output.write(message);
    }



}

File: main/java/com/marsrovers/io/IOHandler.kt
package com.marsrovers.io


import java.util.Scanner

class IOHandler internal constructor(private val scanner: Scanner, private val output: Output) {

    fun readWithMessage(readingMessage: String): String {
        output.write(readingMessage)
        return scanner.nextLine()
    }

    fun write(message: String) {
        output.write(message)
    }


}
